### PR TITLE
Add staging GCR repo for Kubernetes CI images

### PR DESCRIPTION
**Pull request by ameukam**
Add a GCR staging repo that will host images built by
ci-kubernetes-build jobs. It will help migrate those jobs
to k8s-infra-prow-build-cluster.

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

**Comment by justaugustus** [comment]
/retitle Add staging GCR repo for Kubernetes CI images

File: groups/sig-release/groups.yaml
Comment (justaugustus) [inline groups/sig-release/groups.yaml:116]:
```suggestion
      - k8s-infra-release-editors@kubernetes.io
```

File: groups/sig-release/groups.yaml
Comment (justaugustus) [inline groups/sig-release/groups.yaml:121]: Ideally, this is instead:
- IAM for Test Infra folks
- IAM for WG K8s Infra Leads

File: k8s.gcr.io/images/k8s-staging-ci-images/OWNERS
Comment (justaugustus) [inline k8s.gcr.io/images/k8s-staging-ci-images/OWNERS:9]:
```suggestion
  - release-engineering-approvers
  - sig-testing-leads
  - wg-k8s-infra-leads
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
Some nits on the approvers/reviewers/IAM in its current state.

Question to @spiffxp (repost from https://github.com/kubernetes/test-infra/pull/19574/files#r508025503):

Any reason to not use GCR in the `k8s-release` project: https://github.com/kubernetes/k8s.io/pull/1110, https://github.com/kubernetes/k8s.io/issues/846#issuecomment-668795671

The intent of that project was to use it for CI artifacts.
(Right now, the `gs://k8s-release-dev` and `gs://k8s-release-pull` buckets live there.)

File: groups/sig-release/groups.yaml
Comment (spiffxp) [reply groups/sig-release/groups.yaml:121]: I would recommend `k8s-infra-prow-oncall@kubernetes.io` instead
```suggestion
      - k8s-infra-prow-oncall@kubernetes.io
      - ameukam@gmail.com
```

I don't think @listx needs to be included here.  @ameukam I would support you moving to k8s-infra-prow-oncall if you're interested

**Review by spiffxp — APPROVED** [review]
> Any reason to not use GCR in the k8s-release project: #1110, #846 (comment)

- avoiding special cases for the pile of bash that maintains all this; k8s-artifacts-prod and the staging projects are the only projects that have populated GCR repositories and I think it would be better to keep it that way
- GCR repos match project id, so it would have to be gcr.io/k8s-release; I think that's not quite an analogue to kubernetes-ci-images, and might be confusing to see non-releases there

/approve
/lgtm

**Comment by ameukam** [comment]
@spiffxp Very interested in being part of this group. Always happy to give some help. :-)

**Comment by justaugustus** [comment]
Okay, let's party!
/lgtm
/approve
/hold cancel

**Comment by spiffxp** [comment]
This was created to support https://github.com/kubernetes/test-infra/issues/19483

**Comment by ameukam** [comment]
> @ameukam Do we still need `k8s.gcr.io/images/k8s-staging-ci-images/images.yaml` today? It has been two years and it is still empty.
> 
> I'm working on applying #4277 to all the manifests.
> 
> ```
>  REDACTED  MCW0CDP3YY  tmp  client  latest  $  gcrane ls us.gcr.io/k8s-ci-images
> 2022/09/30 14:53:21 No matching credentials were found for "us.gcr.io"
> Error: GET https://us.gcr.io/v2/token?scope=repository%3Ak8s-ci-images%3Apull&service=us.gcr.io: UNKNOWN: Project 'project:k8s-ci-images' not found or deleted.
>  REDACTED  MCW0CDP3YY  tmp  client  latest  ERROR  $  gcrane ls asia.gcr.io/k8s-ci-images
> 2022/09/30 14:59:20 No matching credentials were found for "asia.gcr.io"
> Error: GET https://asia.gcr.io/v2/token?scope=repository%3Ak8s-ci-images%3Apull&service=asia.gcr.io: UNKNOWN: Project 'project:k8s-ci-images' not found or deleted.
>  REDACTED  MCW0CDP3YY  tmp  client  latest  ERROR  $  gcrane ls eu.gcr.io/k8s-ci-images
> 2022/09/30 14:59:27 No matching credentials were found for "eu.gcr.io"
> Error: GET https://eu.gcr.io/v2/token?scope=repository%3Ak8s-ci-images%3Apull&service=eu.gcr.io: UNKNOWN: Project 'project:k8s-ci-images' not found or deleted.
> ```

This project is used to build nightly k8s images used by order prowjobs. There is no image promotion for this project.

**Pull request merged**
